src/types.ts

```diff
@@ -573,20 +574,37 @@ export class Signature {
   cachedFunctionTarget: FunctionTarget | null = null;
   /** Respective function type. */
   type: Type;
+  /** The program that created this signature. */
+  program: Program;
 
   /** Constructs a new signature. */
   constructor(
+    program: Program,
     parameterTypes: Type[] | null = null,
     returnType: Type | null = null,
-    thisType: Type | null = null
+    thisType: Type | null = null,
```

Comment:
nit: trailing comma

Reply:
Ah. This is a personal style choice. Sorry about that.

Reply:
All set.